### PR TITLE
feat: Use `androidx` namespace

### LINKED ISSUE #43: Use `androidx` dependencies
Library consumers won't have to use a Jettifier.

### PR DESCRIPTION
**Pull request by wzieba**
Closes #43

### CODE REVIEW

File: .github/workflows/readme.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ jobs:
         run: ./gradlew :parsely:assembleDebug
       - name: Build example app
         run: ./gradlew :example:assembleDebug
+      - name: Android Lint
+        run: ./gradlew lint
```
Comment (wzieba) [inline .github/workflows/readme.yml:22]: Lint errors has been fixed so we can add this check

File: example/build.gradle [hunk 2]
```diff
@@ -21,12 +20,5 @@ android {
...
 dependencies {
     implementation project(':parsely')
-    implementation 'com.android.support:appcompat-v7:28.0.0'
-    implementation 'com.android.support:customtabs:28.0.0'
-    implementation 'com.android.support:support-v4:28.0.0'
-    implementation 'com.android.support:support-media-compat:28.0.0'
-    implementation 'com.android.support:design:28.0.0'
-    implementation 'org.codehaus.jackson:jackson-mapper-lgpl:1.9.13'
-    implementation 'org.codehaus.jackson:jackson-core-lgpl:1.9.13'
-    implementation 'com.google.android.gms:play-services-ads:17.1.3'
```
Comment (wzieba) [inline example/build.gradle:31]: Unused dependencies

File: example/src/main/java/com/example/MainActivity.java [hunk 3]
```diff
@@ -76,12 +76,6 @@ public void run() {
 
     }
 
-    @Override
-    public boolean onCreateOptionsMenu(Menu menu) {
-        getMenuInflater().inflate(R.menu.main, menu);
-        return true;
-    }
```
Comment (wzieba) [inline example/src/main/java/com/example/MainActivity.java:83]: The menu was not used

**Review by ParaskP7 — COMMENTED** [review]
👋 @wzieba !

I reviewed and testing this PR. In doing so, I left a couple of questions (❓) for you to reply to. Let me know what you think and we can then progress with this work.

PS: This PR is about the usage and update to `AndroidX`. On that aspect everything looks okay. But, since you also made few other changes, Lint on CI and `targetSdkVersion` related, I took this opportunity to comment on those too. I hope my comments will help you! 🙏

File: example/build.gradle [hunk 4]
```diff
@@ -3,12 +3,11 @@ plugins {
 }
 
 android {
-    compileSdkVersion 28
+    compileSdkVersion 32
 
     defaultConfig {
         applicationId "com.parsely.example.parselyexample"
         minSdkVersion 21
-        targetSdkVersion 28
```
[the remark this answers is not in the document]
Comment (wzieba) [reply example/build.gradle:11]: Thanks, good point! Yes, let's specify `targetSdkVersion` to `32`. I went too far with removing configuration code.

File: .github/workflows/readme.yml (hunk 1 above)
Comment (wzieba) [reply .github/workflows/readme.yml:22]: Good point, there's no reason to lint all build types!

File: example/build.gradle (hunk 4 above)
[the remark this answers is not in the document]
Comment (wzieba) [reply example/build.gradle:11]: Additionally, due to the permission issue you've pointed out, I've reverted my previous change of `minSdk` from 21 to 23 (as it was before).

[the remark this answers is not in the document]
Comment (ParaskP7) [reply example/build.gradle:11]: Great, nicely done! 🥇

**Pull request merged**